src/main/kotlin/app/revanced/patches/youtube/layout/theme/patch/ThemePatch.kt

```diff
@@ -45,9 +43,68 @@ class ThemePatch : ResourcePatch {
         }
 
         // copies the resource file to change the splash screen color
-        context.copyResources("theme",
-            ResourceUtils.ResourceGroup("values-night-v31", "styles.xml")
-        )
+        context.xmlEditor["res/values/attrs.xml"].use { editor ->
+            with(editor.file) {
+                val resourcesNode = getElementsByTagName("resources").item(0) as Element
+
+                val newElement: Element = createElement("attr")
+                newElement.setAttribute("format", "reference")
+                newElement.setAttribute("name", "splashScreenColor")
+
+                resourcesNode.appendChild(newElement)
+            }
+        }
+        context.xmlEditor["res/values/styles.xml"].use { editor ->
+            with(editor.file) {
+                val resourcesNode = getElementsByTagName("resources").item(0) as Element
+
+                for (i in 0 until resourcesNode.childNodes.length) {
+                    val node = resourcesNode.childNodes.item(i) as? Element ?: continue
+
+                    val newElement: Element = createElement("item")
+                    newElement.setAttribute("name", "splashScreenColor")
```

Comment:
The scope function `apply` can be used for code such as this.

Reply:
A syntax example of this?

Reply:
[Here](https://kotlinlang.org/docs/scope-functions.html#apply) is a dedicated example.

Reply:
Good. Thanks.